ocrd_network/ocrd_network/utils.py

```diff
@@ -110,3 +111,59 @@ def post_to_callback_url(logger, callback_url: str, result_message: OcrdResultMe
     }
     response = requests.post(url=callback_url, headers=headers, json=json_data)
     logger.info(f'Response from callback_url "{response}"')
+
+
+def get_ocrd_workspace_instance(mets_path: str, mets_server_url: str = None) -> Workspace:
+    if mets_server_url:
+        if not is_mets_server_running(mets_server_url=mets_server_url):
+            raise RuntimeError(f'The mets server is not running: {mets_server_url}')
+
+    workspace = Resolver().workspace_from_url(
+        mets_url=mets_path,
+        mets_server_url=mets_server_url
+    )
+    return workspace
+
+
+def get_ocrd_workspace_physical_pages(mets_path: str, mets_server_url: str = None) -> List[str]:
+    physical_pages = get_ocrd_workspace_instance(
+        mets_path=mets_path,
+        mets_server_url=mets_server_url
+    ).mets.physical_pages
+
+    # This conversion is required since
+    # the type of physical pages is not str
+    str_physical_pages = []
+    for physical_page in physical_pages:
+        str_physical_pages.append(f'{physical_page}')
+    return str_physical_pages
+
+
+def is_mets_server_running(mets_server_url: str) -> bool:
+    if mets_server_url.startswith('http://') or mets_server_url.startswith('https://'):
+        headers = {"Content-Type": "application/json"}
+        response = requests.get(url=f'{mets_server_url}/workspace_path', headers=headers)
+    else:
+        unix_url = f'http+unix://{mets_server_url.replace("/", "%2F")}'
+        try:
+            response = requests_unixsocket.Session().get(url=f'{unix_url}/workspace_path')
+        except Exception:
+            return False
+    if response.status_code == 200:
+        return True
+    return False
```

Comment:
A `log.debug` of the `response` might be useful here, since `/workspace_path` should always return 200 and the path if the server is running.

Reply:
I will have to add an extra parameter `logger` so the caller could pass their own logger type. Is that fine?

Reply:
Sure, but you could also just do `getLogger(logger_name)`, whatever works best in your opinion.